### ISSUE TITLE
Pagination On Groups Is Giving A "Invalid search term entered. GET returned: 404 Invalid page."

### ISSUE THREAD

**Issue by metajar** [ansible/awx#1904]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - UI

##### SUMMARY
Pagination problem in GROUPS tab in Inventories

##### ENVIRONMENT
* AWX version: 1.0.6.8 devel
* AWX install method: docker on linux
* Ansible version: Ansible 2.5.3 
* Operating System: Ubuntu 16.04
* Web Browser: Chrome

##### STEPS TO REPRODUCE

Simply add more than 20 groups to the group tag. Once you have enough groups in place you can try to go through the pages but it will receive a 404 Invalid page message.

##### EXPECTED RESULTS

I would expect to be able to go through the groups.

##### ACTUAL RESULTS

Invalid search term entered. GET returned: 404 Invalid page.

##### ADDITIONAL INFORMATION

```
2018-05-22 21:11:16,428 WARNING  awx.api.generics status 404 received by user admin attempting to access /api/v2/inventories/4/root_groups/ from 192.168.88.25
2018-05-22 21:11:16,433 WARNING  django.request Not Found: /api/v2/inventories/4/root_groups/
2018-05-22 21:11:16,433 WARNING  django.request Not Found: /api/v2/inventories/4/root_groups/
[pid: 29|app: 0|req: 12/59] 192.168.88.25 () {42 vars in 2314 bytes} [Tue May 22 21:11:16 2018] GET /api/v2/inventories/4/root_groups/?page_size=20&order_by=name&page=2 => generated 26 bytes in 63 msecs (HTTP/1.1 404) 8 headers in 235 bytes (1 switches on core 0)
192.168.88.25 - - [22/May/2018:21:11:16 +0000] "GET /api/v2/inventories/4/root_groups/?page_size=20&order_by=name&page=2 HTTP/1.1" 404 26 "http://awx.henry.local/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36"
2018-05-22 21:11:16,494 WARNING  awx.api.generics status 404 received by user admin attempting to access /api/v2/inventories/4/root_groups/ from 192.168.88.25
2018-05-22 21:11:16,499 WARNING  django.request Not Found: /api/v2/inventories/4/root_groups/
2018-05-22 21:11:16,499 WARNING  django.request Not Found: /api/v2/inventories/4/root_groups/
192.168.88.25 - - [22/May/2018:21:11:16 +0000] "GET /api/v2/inventories/4/root_groups/?page_size=20&order_by=name&page=2 HTTP/1.1" 404

```

**Comment by AlanCoding** [comment]
> Simply add more than 20 groups to the group tag.

I assume you meant the group tab.

It looks like it's more complicated than this. I could re-produce the issue, but by:

 - Creating 25 groups
 - adding 5 of those groups as children to another group

This creates the situation where the root groups are _not_ paginated, but the total groups list _is_.

The issue then appears when I navigate to:

`/#/inventories/inventory/50/groups?inventory_search=page_size:20;order_by:name;page:2&group_search=page_size:20;order_by:name;page:1`

This page offers 2 pages, even though the root groups are not paginated.

If I inspect the network traffic, I see some of the confusion to pop up.

![screen shot 2018-05-23 at 11 03 26 am](https://user-images.githubusercontent.com/1385596/40433226-7bc8adb0-5e79-11e8-85ed-087ce723cc48.png)

The first query there was done to the inventory (overall) groups, but the queries related to navigating to the 2nd page were done with the root groups. It seems that the UI is using the page count from the overall groups list, but trying to fetch groups from the root groups list, causing 404 errors.

**Comment by metajar** [comment]
I was coming back to give more detail but that sounds correct. It appears that the root-groups is causing the errors. 

```
192.168.88.25 - - [23/May/2018:17:44:40 +0000] "GET /api/v2/inventories/4/root_groups/?page_size=20&order_by=name&page=2 HTTP/1.1" 404 26 "http://awx.henry.local/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36"
2018-05-23 17:44:40,810 WARNING  awx.api.generics status 404 received by user admin attempting to access /api/v2/inventories/4/root_groups/ from 192.168.88.25
2018-05-23 17:44:40,814 WARNING  django.request Not Found: /api/v2/inventories/4/root_groups/
2018-05-23 17:44:40,814 WARNING  django.request Not Found: /api/v2/inventories/4/root_groups/
```

**Comment by onitake** [comment]
I'm not 100% sure, but I believe this issue is also causing major problems when nested groups are synced from a project inventory. Initially, the group view (correctly) displays a count for all groups found in the inventory, but when trying to navigate to the next page or searching for one of the nested groups, they are suddenly gone and the 404 error is displayed.

I think there is a general misconception here: AWX shouldn't try to reconstruct a group hierarchy in the UI, as a host or a group can be a member of multiple groups at the same time.
Instead, a list of all groups should always be displayed, possibly with a list of the groups the are contained in, for easier navigation. This seems to already be the case, but for some reason it's not working correctly.

**Comment by ryanpetrello** [comment]
@conft @onitake we've just released 1.0.7, which we believe resolves the underlying issue here. You can try it out here: https://github.com/ansible/awx/releases/tag/1.0.7

Let us know if you're still seeing this issue after installing the latest awx - thanks!

**Comment by onitake** [comment]
Thanks, I'd love to try it, but I'm runnining into a schema upgrade problem: #2173
I'll report back when that is fixed.
